deeposlandia/kerastrain.py

```diff
@@ -210,23 +210,56 @@ def add_training_arguments(parser):
     STEPS = args.nb_training_image // args.batch_size
     VAL_STEPS = args.nb_validation_image // args.batch_size
     TEST_STEPS = args.nb_testing_image // args.batch_size
+
+    checkpoint_path = os.path.join(folders["output"], "checkpoints", instance_name)
+    if os.path.isdir(checkpoint_path):
+        checkpoints = os.listdir(checkpoint_path)
+        if len(checkpoints) > 0:
+            model_checkpoint = max(checkpoints)
+            trained_model_epoch = int(model_checkpoint[-5:-3])
+            checkpoint_complete_path = os.path.join(checkpoint_path, model_checkpoint)
+            model.load_weights(checkpoint_complete_path)
+            utils.logger.info(("Model weights have been recovered from {}"
+                               "").format(checkpoint_complete_path))
+        else:
+            utils.logger.info(("No available checkpoint for this configuration. "
+                               "The model will be trained from scratch."))
+            trained_model_epoch = 0
+    else:
+        utils.logger.info(("No available checkpoint for this configuration. "
+                           "The model will be trained from scratch."))
+        trained_model_epoch = 0
+
+    checkpoint_filenames = os.path.join(folders["output"], "checkpoints",
+                                        instance_name, "checkpoint-epoch-{epoch:02d}.h5")
```

Comment:
just a question: you assume with this `02d` that the number of epochs won't be higher than 1000?

Reply:
It won't even be higher than 100. I assumed indeed than the number of epochs won't be larger than this amount, however this may be a strong assumption if we consider larger models, and/or if we decompose epoches in smaller sets.

I may modify `02d` in `03d` for a sake of care.

*As a complementary remark*, note that this name convention is directly related to your first comment about `max(filename)`.